Package 'Coverage' as a higher level API for code coverage

privat opened the pull request:
I propose an independent package that offers a simple API to compute code coverage and to deal with low-level stuff.
While it uses metalink internally, the user is not directly exposed to it.

Currently, compiled methods and sequences of statements (bodies of methods and block) can be tracked.
This is a compromise between efficiency and precision that is good enough precision to track covered paths.

Here is a quick usage:

```st
| collector coverage |
collector := CoverageCollector new. "Instantiate"
collector methods: Point methods, Rectangle methods.  "Configure with the methods to watch."
coverage := collector runOn: [ (1@1 corner: 2@2) center ].  "Setup, execute and teardown."
(coverage methods includes: Point>>#x) >>> true.  "Inspect the results"
(coverage nodes size > 10) >>> true. "Covered paths are also available"
```

Existing usage of metalinks in SUnit and DrTest are also updated to use this new package instead.

privat committed:
New package Coverage for an Higher code coverage API

privat committed:
add CoverageResult>>crown

privat committed:
Update TestRunned and DTTestCoveragePlugin to use the Coverage package

privat commented:
@MarcusDenker can you review?

privat committed:
Try to update baselines to test can pass

privat commented:
I tried to fix the package/baseline issues, but I do not know what I am doing (i should read https://github.com/pharo-open-documentation/pharo-wiki/blob/master/General/Baselines.md )

astares commented:
Can you name the package "Coverage-Core"?
Would be better as we can have additional "Coverage-UI", "Coverage-Tools" or other later.

Ducasse commented:
Cool :)
Eventually I would like to have a branch level coverage.

MarcusDenker committed:
Update BaselineOfDrTests.class.st

Try to fix baseline

MarcusDenker committed:
Update BaselineOfSUnit.class.st

Trying to fix baseline. Not sure yet if this makes sense (we now do not use the BaseLineOfCoverage at all)

MarcusDenker commented:
It loads now! I am not sure if the way we load it is "correct", it seems we do not use the BaseLine at all but instead add the package the baseline of SUnit and DrTest.

MarcusDenker reviewed (approved):
Looks good!

privat committed:
Spelling and fix bugs

privat committed:
Add more tests

privat commented:
Pushed some improvements (and a bugfix)
